qiskit/opflow/gradients/circuit_qfis/lin_comb_full.py

```diff
@@ -179,15 +199,38 @@ def convert(
 
                 # Compute −4 * Re(〈∂kψ|ψ〉〈ψ|∂lψ〉)
                 def phase_fix_combo_fn(x):
-                    return 4 * (-0.5) * (x[0] * np.conjugate(x[1]) + x[1] * np.conjugate(x[0]))
-
-                phase_fix = ListOp(
-                    [phase_fix_states[i], phase_fix_states[j]], combo_fn=phase_fix_combo_fn
-                )
-                # Add the phase fix quantities to the entries of the QFI
-                # Get 4 * Re[〈∂kψ|∂lψ〉−〈∂kψ|ψ〉〈ψ|∂lψ〉]
-                qfi_ops += [SummedOp(qfi_op) + phase_fix]
+                    return -4 * np.real(x[0] * np.conjugate(x[1]))
+
+                if phase_fix:
+                    phase_fix_op = ListOp(
+                        [phase_fix_states[i], phase_fix_states[j]], combo_fn=phase_fix_combo_fn
+                    )
+                    # Add the phase fix quantities to the entries of the QFI
+                    # Get 4 * Re[〈∂kψ|∂lψ〉−〈∂kψ|ψ〉〈ψ|∂lψ〉]
+                    qfi_ops += [SummedOp(qfi_op) + phase_fix_op]
+                else:
+                    qfi_ops += [SummedOp(qfi_op)]
 
             qfi_operators.append(ListOp(qfi_ops))
-        # Return the full QFI
-        return ListOp(qfi_operators, combo_fn=triu_to_dense)
+
+        # Return estimate of the full QFI -- A QFI is by definition positive semi-definite.
+        return ListOp(qfi_operators, combo_fn=lambda x: check_and_realpart(triu_to_dense(x)))
+
+
+def check_and_realpart(x: Union[List[float], np.ndarray]) -> np.ndarray:
+    """
+    Check for non-negligible imaginary values and remove negligible imaginary parts from x.
+    Args:
+        x: vector or matrix
+    Returns:
+        Real part of x
+    Raises:
+        ValueError: If ``x`` has non-negligible imaginary components
+
+    """
+    if np.any([[np.abs(np.imag(x_item)) > 1e-8 for x_item in x_row] for x_row in x]):
```

Comment:
Can we avoid hardcoding `1e-8`?

Reply:
I don't see a reason why we should give the user the possibility to change this checking value. It would only create redundancy.

Reply:
So let's at least make it a variable within this code block with a nice name.